zebrad/tests/acceptance.rs

```diff
@@ -1938,28 +1938,30 @@ fn lightwalletd_integration_test(test_type: TestType) -> Result<()> {
     //
     // If incompletely upgraded states get written to the CI cache,
     // change DATABASE_FORMAT_UPGRADE_IS_LONG to true.
-    if test_type.launches_lightwalletd() && !DATABASE_FORMAT_UPGRADE_IS_LONG {
-        tracing::info!(
-            ?test_type,
-            ?zebra_rpc_address,
-            "waiting for zebrad to open its RPC port..."
-        );
-        wait_for_state_version_upgrade(
-            &mut zebrad,
-            &state_version_message,
-            state_database_format_version_in_code(),
-            [format!(
-                "Opened RPC endpoint at {}",
-                zebra_rpc_address.expect("lightwalletd test must have RPC port")
-            )],
-        )?;
-    } else {
-        wait_for_state_version_upgrade(
-            &mut zebrad,
-            &state_version_message,
-            state_database_format_version_in_code(),
-            None,
-        )?;
+    if !DATABASE_FORMAT_UPGRADE_IS_LONG {
+        if test_type.launches_lightwalletd() {
+            tracing::info!(
+                ?test_type,
+                ?zebra_rpc_address,
+                "waiting for zebrad to open its RPC port..."
+            );
+            wait_for_state_version_upgrade(
+                &mut zebrad,
+                &state_version_message,
+                state_database_format_version_in_code(),
+                [format!(
+                    "Opened RPC endpoint at {}",
+                    zebra_rpc_address.expect("lightwalletd test must have RPC port")
+                )],
+            )?;
+        } else {
+            wait_for_state_version_upgrade(
+                &mut zebrad,
+                &state_version_message,
+                state_database_format_version_in_code(),
+                None,
+            )?;
+        }
```

Comment:
This fixes what seems to be a bug; if the database upgrade is long but the test does not launches lightwalletd, then we still shouldn't wait for the state version upgrade.

Reply:
We shouldn't need to wait for the format upgrade before syncing lwd in the future as there are no plans to extend lightwalletd. It must've been added so the test would wait for the format upgrade that added subtrees and with the expectation that there would be more format upgrades like that one.

Reply:
From what I understand we always need to wait for the database upgrade because this test is used to generated cached images and we don't want to generate half-upgraded cached images. I think we probably need to clean up these tests, I can create a ticket for that.